dashboard/src/components/TreeListingPage/TreeListingPage.tsx

```diff
@@ -135,8 +206,41 @@ const TreeListingPage = ({ inputFilter }: ITreeListingPage): JSX.Element => {
     />
   );
 
+  const InputTime = ({
+    field,
+    fieldError,
+  }: {
+    field: ControllerRenderProps<z.infer<typeof InputTimeSchema>, 'time'>;
+    fieldError: FieldError | undefined;
+  }): JSX.Element => {
+    const className = useMemo(
+      () =>
+        `border-${fieldError ? 'red' : 'gray'} mx-[10px] flex w-[100px] flex-1 rounded border`,
```

Comment:
`border-${fieldError ? 'red' : 'gray'}` 

will bug the tailwind compiler, remember that it interprets the test value not the javascript 

https://tailwindcss.com/docs/content-configuration#dynamic-class-names

Reply:
use the `cn` utils for it